### DISCUSSION TITLE
Cache Related Question

### DISCUSSION THREAD

**Discussion by deep-codepaper** [spatie/laravel-permission#2497] [Q&A] [answered]
Is it considered a good idea to implement caching for permissions while syncing data to roles?

**Allow me to provide an example.** 

Suppose I've added some permissions but forgot to include my custom guard_name. In response, I make changes to the permissions via a SQL query in the database, updating the guard_name for all permissions. However, when I attempt to syncPermissions, an error arises because it appears that there are no permissions associated with my custom guard_name, possibly due to caching.

I would appreciate your input on whether this approach is acceptable to everyone.

**Comment by drbyte** [comment, marked as answer]
Since you bypassed the internal api by running direct db queries, you need to manually clear the cache. 
See https://spatie.be/docs/laravel-permission/v5/advanced-usage/cache
